lib/liquid/context.rb

```diff
@@ -27,7 +27,7 @@ def initialize(environments = {}, outer_scope = {}, registers = {}, rethrow_erro
       @this_stack_used = false
 
       if rethrow_errors
-        self.exception_handler = ->(e) { raise }
+        self.exception_handler = ->(e) { raise e }
```

Comment:
I don't believe adding the `e` changes the behavior. Calling `raise` inside a `rescue` block re-raises the rescued error. For example:

```ruby
begin; raise ArgumentError, 'test'; rescue; raise; end
# and
begin; raise ArgumentError, 'test'; rescue => e; raise e; end
# both result in => ArgumentError: test
```